sycl/unittests/scheduler/BlockedCommands.cpp

```diff
@@ -45,3 +46,87 @@ TEST_F(SchedulerTest, BlockedCommands) {
               Res.MResult == detail::EnqueueResultT::SyclEnqueueSuccess)
       << "The command is expected to be successfully enqueued.\n";
 }
+
+TEST_F(SchedulerTest, DontEnqueueDepsIfOneOfThemIsBlocked) {
+  MockCommand A(detail::getSyclObjImpl(MQueue));
+  A.MEnqueueStatus = detail::EnqueueResultT::SyclEnqueueReady;
+  A.MIsBlockable = true;
+  A.MRetVal = CL_SUCCESS;
+
+  MockCommand B(detail::getSyclObjImpl(MQueue));
+  B.MEnqueueStatus = detail::EnqueueResultT::SyclEnqueueReady;
+  B.MIsBlockable = true;
+  B.MRetVal = CL_SUCCESS;
+
+  MockCommand C(detail::getSyclObjImpl(MQueue));
+  C.MEnqueueStatus = detail::EnqueueResultT::SyclEnqueueBlocked;
+  C.MIsBlockable = true;
+
+  MockCommand D(detail::getSyclObjImpl(MQueue));
+  D.MEnqueueStatus = detail::EnqueueResultT::SyclEnqueueReady;
+  D.MIsBlockable = true;
+  D.MRetVal = CL_SUCCESS;
+
+  addEdge(&A, &B, nullptr);
+  addEdge(&A, &C, nullptr);
+  addEdge(&A, &D, nullptr);
+
+  // We have such a graph:
+  //
+  //     A
+  //   / | \
+  //  B  C  D
+  //
+  // If C is blocked, we should not try to enqueue D.
```

Comment:
Do I understand it correctly that `A` depends on `B`, `C` and `D`? Why can't we enqueue `D` then?

Reply:
We discussed it with @romanovvlad and AFAIU, in the current design there are no benefits of including `D`, as it is likely to be blocked as well.